notebook-lessons/oss-module-01-git.ipynb

```diff
@@ -2709,6 +2712,246 @@
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4dd7105f",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "# 6: Git Commit Signing and DCO Signoff\n",
+    "\n",
+    "<img src=\"../img/icon-gears.png\" width=\"200\" />\n",
+    "\n",
+    "In professional and enterprise environments, you may encounter requirements for **signed commits** and **DCO signoff**. These are important security and legal compliance measures that help establish trust and legal clarity in collaborative projects."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c62e3d6b",
+   "metadata": {},
+   "source": [
+    "## 6.1: Developer Certificate of Origin (DCO) Signoff\n",
+    "\n",
+    "### What is `git commit --signoff`?\n",
+    "\n",
+    "The `--signoff` flag adds a \"Signed-off-by\" line to your commit message, which is a **requirement in some projects** - most notably the Linux kernel and official IBM projects.\n",
+    "\n",
+    "This practice originated in response to the [SCO lawsuit(s) versus IBM](https://en.wikipedia.org/wiki/SCO%E2%80%93Linux_disputes) and helps to solve concerns around copyright infringement. The sign-off effectively serves as a **Developer Certificate of Origin (DCO)**, asserting that:\n",
+    "\n",
+    "- ✅ You wrote the code yourself, **or**\n",
+    "- ✅ You have the right to submit it under an appropriate open-source license, **or**\n",
+    "- ✅ Someone else provided it to you under such terms\n",
+    "\n",
+    "### Using DCO Signoff\n",
+    "\n",
+    "To sign off on a commit, use the `--signoff` or `-s` flag:\n",
+    "\n",
+    "```bash\n",
+    "git commit -s -m \"Fix bug in user authentication\"\n",
+    "```\n",
+    "\n",
+    "This will add a line like this to your commit message:\n",
+    "```\n",
+    "Signed-off-by: Your Name <your.email@example.com>\n",
+    "```\n",
+    "\n",
+    "### Setting Up Automatic DCO Signoff\n",
+    "\n",
+    "For projects that require DCO signoff, you can create an alias to make this automatic:\n",
+    "\n",
+    "```bash\n",
+    "# Create an alias that always includes signoff\n",
+    "git config --global alias.ci 'commit --signoff'\n",
+    "\n",
+    "# Now you can use:\n",
+    "git commit -m \"Your commit message\"\n",
+    "```\n",
+    "\n",
+    "### Fixing Missing DCO Signoff\n",
+    "\n",
+    "If you forgot to sign off on past commits, you can retroactively apply the sign-off:\n",
+    "\n",
+    "```bash\n",
+    "# Note: replace the `#` with however many commits you want to go back and update\n",
+    "git rebase -i HEAD~# --signoff\n",
+    "\n",
+    "# Then run:\n",
+    "git push --force-with-lease origin main\n",
+    "```\n",
+    "\n",
+    "> **Warning**: This will replace your commit messages if you're not careful! Always backup your work before force-pushing."
```

Comment:
I recommend omitting this section.  In collaborative workflows, there are many cases where (in general) force-pushing to a published branch can cause irreparable harm to the team.  You might think that it won't matter if you do it in an un-merged branch on your own fork, but it's not uncommon for other team members to base their changes off of your un-merged branch.  

So the situation for fixing missing signoffs is complicated, depends on your team's practices, and it's and best resolved by asking your project manager.  We shouldn't be overly-proscriptive in our guide.